### ISSUE TITLE
Namespace and Homepage are not generated 

### ISSUE THREAD

**Issue by aminya** [userscripters/generate-headers#119]
In the default programmatic example, the namespace and homepage are not generated. I tried changing the first input to `tampermonkey`, but it doesn't change the result

https://stackapps.com/q/9088/92224

Also, the import seems wrong in Stackoverflow. It should be
```
import { generate } from "@userscripters/generate-headers/dist/generate"
```

**Comment by Oaphi** [comment]
Hmm, can you share what's the package.json `name` and `homepage` fields look like in your case, @aminya? The `@namespace` header is a common header, so it is supposed to be generated for all 3 userscript managers supported (and in fact it is, at least for all the userscripts of the org that use the package - I wonder if there is something specific about your scope) As for the programmatic import - yikes, indeed, my bad, typo'd (also will reexport in the next version)

**Comment by aminya** [comment]
I have uploaded the code here:
https://github.com/aminya/jira-issue-navigate/pull/2

The package.json is just a standard file:
https://github.com/aminya/jira-issue-navigate/blob/header-generator/package.json

**Comment by Oaphi** [comment]
Ah, I see what's up here, @aminya  - for many of the headers, by design, the package relies on the information from package.json. In the example you provided, the field `homepage` is missing, hence no `@homepage` header generated (I am open to suggestions as to what field to default to if you'd like that to be added - `repository` maybe?). 

As for the `@namespace` header, the logic of the package is to use NPM's notion of `scope`, which is the first part of the `@scope/packagename`. Your package.json only contains an unscoped name (`jira-issue-navigate`), hence no `@namespace` header generated. I see two possible outcomes here: if you don't have a problem with scoping the package, then just prefix your namespace, *or* we can expose something like `scope` on both CLI and programmatic export.

**Comment by aminya** [comment]
> the field homepage is missing, hence no `@homepage` header generated (I am open to suggestions as to what field to default to if you'd like that to be added - repository maybe?).

Oh, I see. Yes, using the repository will be a very good default! 

> As for the `@namespace` header, the logic of the package is to use NPM's notion of scope, which is the first part of the @scope/packagename. Your package.json only contains an unscoped name (jira-issue-navigate), hence no `@namespace` header generated. I see two possible outcomes here: if you don't have a problem with scoping the package, then just prefix your namespace, or we can expose something like scope on both CLI and programmatic export.

I don't know if namespace is needed, but adding the field doesn't generate it. I remember greasyfork complaining about not having a namespace. 

BTW, I noticed another bug in the StackOverflow post. `package` should become `packagePath`. It also specifies that the `namespace` defaults to `name`, but it doesn't.

**Comment by Oaphi** [comment]
@aminya for now, I've decided to add `--homepage` and `--namespace` options to allow for overriding of them in v3.0.0 (be careful, though, as I've been planning on upgrading to native ESM for a while and finally did this for that release) - which is released. I'll add defaulting to the `repository` field in the next minor release (likely, tomorrow).

As for the namespace, I think I might not have been clear: the package uses the `name` field from package.json to determine both the namespace and name (similarly to NPM). For example, `@userscripters/my-userscript` will generate `// @namespace userscripters` and `// @name my-userscript`  respectively. Adding a custom field to `package.json` will not work (although I am open to adding a custom config field in package.json in a future version).

Regarding the documentation, the `namespace` does not default to name, it just uses the `name` field from package.json with rules defined as described in the previous paragraph. However, it could be made clearer, I'll update the docs to make sure it cannot be interpreted as if `name` was the default.

And, finally, yeah, `package` should be `packagePath` - I try to keep the StackApps post up-to-date, but it's not easy to remember all the changes :) However, I have to note that the CLI option is still `--package`, nothing has changed in thar regards (come to think of it, makes sense to add a `--package-path` alias to it, though).

**Comment by aminya** [comment]
I just tested the new arguments. They worked as expected, however, the license was not added to the header.

**Comment by double-beep** [comment]
@aminya should be fixed in v3.4.2. Thanks for the report!
